pg_shard.c

```diff
@@ -393,7 +393,9 @@ ErrorIfQueryNotSupported(Query *queryTree)
 	bool hasValuesScan = false;
 	uint32 queryTableCount = 0;
 	bool hasNonConstTargetEntryExprs = false;
+	bool hasNonConstQualExprs = false;
 	bool specifiesPartitionValue = false;
+	Node *whereClause = NULL;
```

Comment:
This variable can have its scope reduced… please put it inside the `if (commandType == CMD_INSERT || commandType == CMD_UPDATE ||…` block.